src/shared_utilities/general_utils.jl

```diff
@@ -16,3 +16,21 @@ function searchsortednearest(a, x)
         return abs(a[i] - x) < abs(a[i - 1] - x) ? i : i - 1
     end
 end
+
+
+"""
+"""
+function linear_interpolation(indep_vars, dep_vars, indep_value)
+    N = length(indep_vars)
+    id = searchsortedfirst(indep_vars, indep_value)
+    if id == 1
+        dep_vars[id]
+    elseif id == N
```

Comment:
edge case I ran across, I think the time case has an @assert that `min(indep_vars) <= indep_value <= max(indep_vars)`